modules/library/referencing/src/main/java/org/geotools/referencing/operation/matrix/XMatrix.java

```diff
@@ -112,6 +136,48 @@
     void invert() throws SingularMatrixException;
 
     /**
+     * Set to the inverse of the provided matrix.
+     * 
+     * @param matrix The matrix that is to be inverted. Not modified.
+     * @throws SingularMatrixException if this matrix is not invertible.
+     */
+    void invert( Matrix matrix ) throws SingularMatrixException;
```

Comment:
A bit confused here, why are all these methods in XMatrix now? I guess because you have to make sure code expecting GeneralMatrix works also with MatrixN (N=1,2,3,4?)